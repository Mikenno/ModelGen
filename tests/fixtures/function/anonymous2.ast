Module
- Call
-- Name print
-- Call
--- Name map
--- Function
---- Invalid
---- Tuple
----- Name a
---- Return
----- BinOpMul
------ Name a
------ Name a
--- Call
---- Name range
---- Integer 5
- Call
-- Name print
-- Call
--- Name map
--- Function
---- Invalid
---- Tuple
----- Name a
---- Return
----- BinOpMul
------ Name a
------ Name a
--- Call
---- Name range
---- Integer 5
- Call
-- Name print
- Call
-- Name print
-- Call
--- Name map
--- Function
---- Invalid
---- Tuple
----- Name a
----- Name b
---- Return
----- BinOpMul
------ Name a
------ Name b
--- Call
---- Name range
---- Integer 5
--- Call
---- Name range
---- Integer 10
---- Integer 15
- Call
-- Name print

-- Call
--- Name map
--- Function
---- Invalid
---- Tuple
----- Name a
----- Name b
---- Return
----- BinOpMul
------ Name a
------ Name b
--- Call
---- Name range
---- Integer 5
--- Call
---- Name range
---- Integer 10
---- Integer 15
- Call
-- Name print

- Assign
-- Name f
-- Function
--- Invalid
--- Tuple
--- Return
---- Integer 1
- Call
-- Name print
-- Call
--- Name f
-- Call
--- Function
---- Invalid
---- Tuple
---- Return
----- Integer 1

- Assign
-- Name f
-- Function
--- Invalid
--- Tuple
---- Name a
--- Return
---- Call
----- Name print
----- Name a
- Call
-- Name f
-- Integer 123
- Call
-- Function
--- Invalid
--- Tuple
---- Name a
--- Return
---- Call
----- Name print
----- Name a
-- Integer 456
- Call
-- Name print

- Assign
-- Tuple
--- Name add
--- Name square
-- Tuple
--- Function
---- Invalid
---- Tuple
----- Name a
------ Name float
----- Name b
------ Name float
---- Return
----- BinOpAdd
------ Name a
------ Name b
--- Function
---- Invalid
---- Tuple
----- Name a
------ Name float
---- Return
----- BinOpMul
------ Name a
------ Name a

- Call
-- Name print
-- Call
--- Name add
--- Integer 1
--- Integer 1
-- Call
--- Name add
--- Integer 1
--- Integer 2
-- Call
--- Name add
--- Integer 2
--- Integer 1
-- Call
--- Name add
--- Integer 2
--- Integer 2
- Call
-- Name print
-- Call
--- Name square
--- Integer 1
-- Call
--- Name square
--- Integer 2
-- Call
--- Name square
--- Integer 3
-- Call
--- Name square
--- Integer 4