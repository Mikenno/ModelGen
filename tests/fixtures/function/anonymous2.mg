
print(map((func(a) return a * a), range(5)))
print(map(a -> a * a, range(5)))
print()

print(map((func(a, b) return a * b), range(5), range(10, 15)))
print(map((a, b) -> a * b, range(5), range(10, 15)))
print()

f = () -> 1
print(f(), (() -> 1)())

f = (a) -> print(a)
f(123)
((a) -> print(a))(456)
print()

add, square = (a: float, b: float) -> a + b, a: float -> a * a

print(add(1, 1), add(1, 2), add(2, 1), add(2, 2))
print(square(1), square(2), square(3), square(4))
