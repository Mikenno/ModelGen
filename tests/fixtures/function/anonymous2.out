[0, 1, 4, 9, 16]
[0, 1, 4, 9, 16]

[0, 11, 24, 39, 56]
[0, 11, 24, 39, 56]

1 1
123
456

2 3 3 4
1 4 9 16
